### PR TITLE
[BUG] Fix Prophet with logistic growth #1079

### LINKED ISSUE #1079: [BUG] Prophet with logistic growth does not work
**Describe the bug**
It is not compatible with the case that the growth parameter of prophet is equal to ‘logistic‘

**To Reproduce**
```python

from sktime.datasets import load_airline
y = load_airline()
y = y.to_timestamp(freq="M")
y_train, y_test = temporal_train_test_split(y, test_size=36)

y_train_df = pd.DataFrame(y_train,index = y_train.index,columns=['y'])
y_train_df["cap"] = max(y_train_df["y"])
y_train_df['floor'] = 0

forecaster = Prophet(
    growth = 'logistic'
)

forecaster.fit(y_train_df)
```

**Expected behavior**

The growth parameter compatible with prophet is equal to ‘logistic‘
**Additional context**

**Versions**
<details>

from sktime import show_versions; show_versions()

System:
    python: 3.8.3 (default, Jul  2 2020, 11:26:31)  [Clang 10.0.0 ]
executable: /Users/baixiaotiao/opt/anaconda3/bin/python
   machine: macOS-10.16-x86_64-i386-64bit

Python dependencies:
          pip: 20.1.1
   setuptools: 49.2.0.post20200714
      sklearn: 0.24.2
       sktime: 0.6.1
  statsmodels: 0.12.2
        numpy: 1.20.3
        scipy: 1.5.0
       Cython: 0.29.21
       pandas: 1.2.5
   matplotlib: 3.2.2
       joblib: 0.16.0
        numba: 0.50.1
     pmdarima: 1.8.2
      tsfresh: None

</details>

### PR DESCRIPTION
**Pull request by k1m190r**
#### Reference Issues/PRs
Fixes #1079 and #2444
Copy and paste of #1868 

#### What does this implement/fix? Explain your changes.
Adds the possibility to define the required "cap" and "floor" parameters for Prophet when its growth is set to logistic.

#### Any other comments?
Suggest closing old relevant issues and PRs.

##### For all contributions
- [x] Tests pass locally and [private github-action](https://github.com/k1m190r/sktime/actions/runs/2280446935)
- [x] The PR title starts with [BUG]

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
Looks good to me!

my main blocking change requests would be:

* @aiwalter's comments
* I think we need to do something in case the columns `"cap"` and `"floor"` already exist. An option would be to temporarily rename them to `"cap__n"` etc where `n` is the smallest positive integer that the string isn't already a column name

**Comment by k1m190r** [comment]
@fkiraly if `cap` and `floor`  cols exist in `y` should the values from the columns be used as `growth_cap` and `growth_floor` params and removed from `y`? Since Prophet expects univariate.

File: sktime/forecasting/fbprophet.py [hunk 1]
```diff
@@ -126,6 +126,8 @@ def __init__(
         add_country_holidays=None,
         # Args of fbprophet
         growth="linear",
+        growth_floor=0.0,
+        growth_cap=None,
```
[the remark this answers is not in the document]
Comment (k1m190r) [reply sktime/forecasting/fbprophet.py:137]: In `Prophet` judging by the [code here ](https://github.com/facebook/prophet/blob/b7d5ec448177b7ac2abf25cc3259cb55e4c44069/python/prophet/forecaster.py#L321)`floor` default is 0.0, `cap` has no default and must be specified.

docstring added

File: sktime/forecasting/fbprophet.py [hunk 2]
```diff
@@ -142,7 +151,7 @@ def __init__(
         uncertainty_samples=1000,
         stan_backend=None,
         verbose=0,
-        interval_width=0,
+        interval_width=0.80,
```
Comment (k1m190r) [inline sktime/forecasting/fbprophet.py:154]: Synced this [default parameter ](https://github.com/facebook/prophet/blob/b7d5ec448177b7ac2abf25cc3259cb55e4c44069/python/prophet/forecaster.py#L93)value with Prophet.

**Comment by k1m190r** [comment]
Thank you for the corrections gents. I have noticed another [BUG] while implementing this PR. Will open new issue #2621.

**Comment by fkiraly** [comment]
>@fkiraly if `cap` and `floor`  cols exist in `y` should the values from the columns be used as `growth_cap` and `growth_floor` params and removed from `y`? Since Prophet expects univariate.

No, I think in no case should values from `y` columns be used as parameters. That would break the unified interface.

File: sktime/forecasting/fbprophet.py [hunk 3]
```diff
@@ -38,6 +38,13 @@ class Prophet(_ProphetAdapter):
     growth: str, default="linear"
         String 'linear' or 'logistic' to specify a linear or logistic
         trend.
+    growth_floor: float, array like, default=0
+        Growth saturation minimum value. If float floor is constant.
+        If array like - Variable floor (must be same length as y).
+    growth_cap: float, array like
```
Comment (fkiraly) [inline sktime/forecasting/fbprophet.py:44]: should say what the default is

**Comment by fkiraly** [comment]
See #2626 on the `interval_width` param. I think it can be removed?

**Comment by k1m190r** [comment]
> * optional parameter `growth_cap` needs to be indicated as optional in the docstring. Conditionally non-optional parameter `growth_floor` should say it is non-optional if `growth="logistic"` in the docstring.

- [ ] `growth_cap` is conditionally optional if `growth="logistic"` and `growth_floor` optionality implied by default=0?

> * I think array-like `growth_cap` and `growth_floor` do not work? So, docstring should not mention them.

- [x] removed


> * we can't change existing default params without deprecation, but I also think `interval_width` is not used, so I'd leave it or remove it entirely

- [x] reverted to 0

**Comment by fkiraly** [comment]
>  is conditionally optional if growth="logistic" and growth_floor optionality implied by default=0?

Well, docstring needs to make clear that `growth_cap` is usually optional.
Currently, a reader could think setting `growth_cap` is mandatory in all cases.

**Comment by k1m190r** [comment]
```
growth_floor: float, default=0
    Growth saturation minimum value.

growth_cap: float, default=None
    Growth saturation maximum aka carrying capacity.
    Conditionally optional: must be specified if `growth="logistic"`.
```

Is this ok?

**Comment by fkiraly** [comment]
> Is this ok?

Not sure - is `None` not ok for `growth_cap` if `growth="logistic"`?
Would `1` make sense as a sensible default?

I would also write "Used only if `growth="logistic"`" for `growth_floor`.

**Comment by fkiraly** [comment]
@k1m190r, are you still working on this?

**Comment by k1m190r** [comment]
```
    growth: str, default="linear"
        String 'linear' or 'logistic' to specify a linear or logistic
        trend. If 'logistic' specified float for 'growth_cap' must be provided.
    growth_floor: float, default=0
        Growth saturation minimum value.
    growth_cap: float, default=None
        Growth saturation maximum aka carrying capacity.
        Conditionally optional: must be float if 'growth="logistic"'.
```

Ok I think this would be it.
- if `growth="linear"`: both `growth_floor` and growth_cap` are optional.
- if `growth="logistic"`: `growth_floor` is still optional (default=0) but `growth_cap` is not optional and must be non None - float.

**Comment by fkiraly** [comment]
makes sense. I would remove the "conditionally optional" in favour of a clearer formulation, since this does not state what the condition is.

Last line, I would change to:
```
mandatory (float) iff `growth="logistic"`, has no effect and is optional, otherwise (if `growth` is not `"logistic"`)
```

**Comment by k1m190r** [comment]
Done.

```
    growth: str, default="linear"
        String 'linear' or 'logistic' to specify a linear or logistic
        trend. If 'logistic' specified float for 'growth_cap' must be provided.
    growth_floor: float, default=0
        Growth saturation minimum value.
        Used only if  `growth="logistic"`, has no effect otherwise
        (if `growth` is not `"logistic"`).
    growth_cap: float, default=None
        Growth saturation maximum aka carrying capacity.
        Mandatory (float) iff `growth="logistic"`, has no effect and is optional,
        otherwise (if `growth` is not `"logistic"`).
```

**Review by fkiraly — APPROVED** [review]
Thanks!

**Pull request merged**